packages/web3-utils/src/validation.ts

```diff
@@ -390,3 +407,100 @@ export function isTopicInBloom(bloom: string, topic: string): boolean {
 
 	return isInBloom(bloom, topic);
 }
+
+export const isBlockNumber = (value: Uint): boolean =>
+	isHexStrict(value) && value.substr(0, 1) !== '-';
+
+/**
+ * Returns true if the given blockNumber is 'latest', 'pending', or 'earliest.
+ */
+export const isBlockTag = (value: string) =>
+	BlockTags.LATEST === value || BlockTags.PENDING === value || BlockTags.EARLIEST === value;
+
+/**
+ * Returns true if given value is valid hex string and not negative, or is a valid BlockTag
+ */
+export const isBlockNumberOrTag = (value: BlockNumberOrTag) =>
+	(isHexStrict(value) && !value.startsWith('-')) || isBlockTag(value);
+
+export const validateBlockNumberOrTag = (value: BlockNumberOrTag) => {
+	if (!isBlockNumberOrTag(value)) throw new InvalidBlockNumberOrTagError(value);
+};
+
+export const isHexString8Bytes = (value: HexString8Bytes) =>
+	isHexStrict(value) && value.length === 18; // 8 bytes + 0x = 18
+
+export const validateHexString8Bytes = (value: HexString8Bytes) => {
+	if (!isHexString8Bytes(value)) throw new InvalidHexStringError(value, 8);
+};
+
+export const isHexString32Bytes = (value: HexString32Bytes) =>
+	isHexStrict(value) && value.length === 66; // 32 bytes + 0x = 66
+
+export const validateHexString32Bytes = (value: HexString32Bytes) => {
+	if (!isHexString32Bytes(value)) throw new InvalidHexStringError(value, 32);
+};
```

Comment:
Should also validate values without `0x` prefix with a `hasPrefix: boolean` argument